pymc3/model.py

```diff
@@ -102,64 +103,70 @@ def get_named_nodes_and_relations(graph):
     graph - a theano node
 
     Returns:
-    leaf_nodes: A dictionary of name:node pairs, of the named nodes that
-        are also leafs of the graph
-    node_parents: A dictionary of node:set([parents]) pairs. Each key is
+    leafs: A dictionary of name:node pairs, of the named nodes that
+        have no named ancestors in the provided theano graph.
+    descendents: A dictionary of node:set([parents]) pairs. Each key is
         a theano named node, and the corresponding value is the set of
-        theano named nodes that are parents of the node. These parental
-        relations skip unnamed intermediate nodes.
-    node_children: A dictionary of node:set([children]) pairs. Each key
+        theano named nodes that are direct descendents of the node in the
+        supplied ``graph``. These relations skip unnamed intermediate nodes.
+    ancestors: A dictionary of node:set([ancestors]) pairs. Each key
         is a theano named node, and the corresponding value is the set
-        of theano named nodes that are children of the node. These child
-        relations skip unnamed intermediate nodes.
+        of theano named nodes that are direct ancestors in the of the node in
+        the supplied ``graph``. These relations skip unnamed intermediate
+        nodes.
 
     """
     if graph.name is not None:
```

Comment:
This is here because we don't enforce distribution parameters to have a name but we may but we may attempt to get_named_nodes_and_relations from them anyway in `distributions.draw_values`. This means that must take care only to add `graph` to the ancestors and descendents dictionaries if it has a name.

Reply:
I was working on this and actually, it looks like unnamed graphs are skipped directly in `draw_values`. That looks kind of wrong so I'm confused now...

Reply:
In the end, the unnamed nodes weren't skipped so this if else statement is necessary